lib/schemaUtils.js

```diff
@@ -3399,17 +3401,48 @@ module.exports = {
     }, (err, res) => {
       let mismatches = [],
         mismatchObj;
+      const unmatchedSchemaVariableNames = determinedPathVariables.filter((pathVariable) => {
+        return !pathVariable._varMatched;
+      }).map((schemaPathVar) => {
+        return schemaPathVar.key;
+      });
 
       if (err) {
         return callback(err);
       }
 
       // go through required schemaPathVariables, and params that aren't found in the given transaction are errors
-      _.each(schemaPathVariables, (pathVar) => {
+      _.each(schemaPathVariables, (pathVar, index) => {
         if (!_.find(determinedPathVariables, (param) => {
           // only consider variable matching if url path variables is not allowed
           return param.key === pathVar.name && (options.allowUrlPathVarMatching || param._varMatched);
         })) {
+          let reasonCode,
+            reason,
+            suggestedValue,
+            actualValue,
+            currentUnmatchedVariableInTransaction = unmatchedVariablesFromTransaction[index],
+            isInvalidValue = currentUnmatchedVariableInTransaction !== undefined;
+
+          if (unmatchedSchemaVariableNames.length > 0 && isInvalidValue) {
+            reasonCode = 'INVALID_VALUE';
+            reason = `The ${currentUnmatchedVariableInTransaction} path variable does not match with path variable` +
```

Comment:
I think the way this text reads could be improved a bit. Writing `at this position (${unmatchedSchemaVariableNames[index]})` expects us to provide a position index rather than the suggestedValue inside the parenthesis. Could instead be something like:

```
reason = `The ${currentUnmatchedVariableInTransaction} path variable does not match with path variable` +
    ` expected (${unmatchedSchemaVariableNames[index]}) in the schema at this position`;
```